service/controller/v5/resource/chart/update.go

```diff
@@ -51,7 +51,7 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 			return microerror.Mask(err)
 		}
 
-		err = r.helmClient.UpdateReleaseFromTarball(ctx, releaseName, tarballPath, helm.UpdateValueOverrides(values))
+		err = r.helmClient.UpdateReleaseFromTarball(ctx, releaseName, tarballPath, helm.UpdateValueOverrides(values), helm.UpgradeForce(true))
```

Comment:
IMO this is dangerous because [it means could delete the chart and installed it again](https://github.com/helm/helm/issues/4177#issuecomment-407463879). For some charts like nginx controller it would have an impact for customer workloads. Are we sure?

cc/ @puja108

Reply:
Late to the party but agree with @pipo02mix that `upgrade --force` is dangerous for cluster critical apps. @TheoBrigitte Let me know if I can help here.

For me the failed releases is the biggest support problem with chart-operator. We need a solution and @giantswarm/team-batman will be working on chart-operator a lot for the managed apps story.

Reply:
Yeah I agree force upgrade is quite dangerous here and there might be some cases of failed chart which do not even need this.

I feel like the logic for state transition is something that would be needed to implement the force upgrade (or another recovery solution) later on. WDYT if we just go with state transition here ?

Reply:
@TheoBrigitte Thanks, going with the state transition logic for now is a good idea. 

I'll take care of reconciling the chart status as part of the App Catalog story. But we're still discussing how to configure the `upgrade --force` support in the spec PR.